tests/test_mesh.py

```diff
@@ -181,5 +181,21 @@ def runTest(self):
         self.assertTrue('top' in mesh.boundaries)
 
 
+class TestMeshQuadSplit(unittest.TestCase):
+
+    def runTest(self):
+        from docs.examples.ex17 import mesh
+        tri_mesh = mesh.to_trimesh()
+
+        for s in mesh.subdomains:
+            self.assertEqual(np.setdiff1d(*[m.t.T[m.subdomains[s]]
+                                            for m in [mesh, tri_mesh]]).size,
+                             0)
+
+        for b in mesh.boundaries:
+            np.testing.assert_array_equal(*[m.facets.T[m.boundaries[b]]
+                                            for m in [mesh, tri_mesh]])
+
```

Comment:
Here is an additional test that should be resolved:
```python
>>> m = MeshQuad()
>>> m.define_boundary('left', lambda x: x[0] == 0)
>>> m.to_trimesh()
Triangular mesh with 4 vertices and 2 elements.
>>> m.to_trimesh().boundaries
{'left': array([1])}
>>> m.refine()
>>> m.to_trimesh()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/src/scikit-fem/skfem/mesh/mesh2d/mesh_quad.py", line 269, in to_trimesh
    for f in self.facets.T[self.boundaries[k]]])
  File "/src/scikit-fem/skfem/mesh/mesh2d/mesh_quad.py", line 269, in <listcomp>
    for f in self.facets.T[self.boundaries[k]]])
StopIteration
```

Reply:
Ugh.  Is this something to do with #104? That has `MeshQuad` ticked. 

I knew those functional programming people were onto something when they lauded immutability.

Reply:
`m.to_meshtri().boundaries['left'] == [6, 1]`, which isn't sorted which was assumed.  I'll sort.  Is it only `refine` that deranges boundaries?  I'll add the test.

It's also assumed that `.facets` is sorted lexicographically.

Reply:
I've sorted in _splitquads, but an alternative would be to sort in `Mesh._fix_boundaries`. 

https://github.com/kinnala/scikit-fem/blob/b5571dac8bad50aee2da1ad6a1146f488c49e400/skfem/mesh/mesh.py#L144-L145

Reply:
...instead of `.flatten()`

Reply:
> I knew those functional programming people were onto something when they lauded immutability.

Yeah, it's an old mistake that `refine` does not return a new mesh. Sometimes I ran into an issue where I run `m.refine()` and then the respective `InteriorBasis` breaks.

I wonder if we should do something about it or not.

Reply:
> I've sorted in _splitquads, but an alternative would be to sort in `Mesh._fix_boundaries`.
> 
> https://github.com/kinnala/scikit-fem/blob/b5571dac8bad50aee2da1ad6a1146f488c49e400/skfem/mesh/mesh.py#L144-L145

I cannot currently recall a reason why there couldn't be a sort there.

I'll add an issue as a reminder.